src/libstd/dynamic_lib.rs

```diff
@@ -263,6 +266,16 @@ mod dl {
                  symbol: *const libc::c_char) -> *mut libc::c_void;
         fn dlclose(handle: *mut libc::c_void) -> libc::c_int;
     }
+
+    #[cfg(target_os = "openbsd")]
```

Comment:
Yeah I think that `#[link_name]` isn't ready any more and could probably be removed from the above block as well. A linkage of `-ldl`  should probably happen in `rtdeps.rs` anyway